lib/endpoints/vms.js

```diff
@@ -2162,13 +2162,19 @@ function createVm(req, res, next) {
                     error));
                 return;
             } else {
-                req.params.sync = (sync === 'true' ? true : false);
+                req.params.sync = (sync === 'true') ? true : false;
             }
 
             var locality = req.params.locality;
-            if (locality) {
+            var encrypted = req.params.encrypted;
+            if (locality || encrypted) {
                 var metadata = req.params.internal_metadata || {};
-                metadata.locality = JSON.stringify(locality);
+                if (locality) {
+                    metadata.locality = JSON.stringify(locality);
+                }
+                if (encrypted) {
+                    metadata.encrypted = true;
+                }
```

Comment:
Wait, why do we have both **vm.encrypted** and **vm.internal_metadata.encrypted** ? It should only be one or the other (for consistency).

Reply:
I want `vm.encrypted` to be a parameter sent to the createVM workflow so it's very explicit there what are we looking for. Other than that, it's useless for the VM object itself, unless we need to migrate the machine to another server. Hence storing the value into internal_metadata.

I think that it's easier to document `vm.encrypted` as a `POST /vms` parameter, then forget about it completely, since it's something the CN will provide. I don't find necessary to have to store it into the VM object itself, neither have to modify `vmadm` for this.

So, note we haven't got `vm.encrypted` but `req.params.encrypted`, which is what I think will make it easier to use.

Has sense?

Reply:
Can CloudAPI and the workflow parameters use internal_metadata to specify if the encryption is needed? I would like to avoid having magical properties on a vm payload (or vmobject) that are then converted into a different property - as that makes it hard for other tools (e.g. migrate has to check if encryption is enabled in internal_metadata, then go and set vmpayload.encryption property - this shouldn't have to happen - it would be better if it can just pass through the vmobj as it is).